utils/repos.py

```diff
@@ -35,6 +39,20 @@ def parse_repository_metadata(self, base_url):
         metadata = self.download_and_unpack(urljoin(base_url, "repodata/repomd.xml"))
         return {d.get("type"): urljoin(base_url, d.location.get("href")) for d in metadata.data}
 
+    @staticmethod
+    def newest_package(first: Package, second: Package):
+        """
+        Compares the version of the two packages and returns the newest.
+        In case they are identical, the first one is returned
+
+        :param first: First package
+        :param second: Second package
+        :return: the newest of the two
+        """
+        result = rpm_vercmp.vercmp(f"{first.version}-{first.release}",
+                                   f"{second.version}-{second.release}")
+        return second if result < 0 else first
```

Comment:
That looks good to me.

Just an idea on how to implement this differently, in case you like it (but feel free to go with your implementation, it's perfectly fine): instead of a static method, define `Package.__lt__(self, other)`, and where you're calling the static method you can use Python's built-in `min(arg1, arg2)` instead.

Reply:
That sounds like a cleaner (and more object-oriented) approach indeed. I actually didn't know it was possible to add methods to namedtuples. Let me change it then